pkgs/development/tools/pnpm/fetch-deps/default.nix

```diff
@@ -36,15 +36,22 @@
         name = "${pname}-pnpm-deps";
 
         nativeBuildInputs = [
+          cacert
           jq
           moreutils
           pnpm
-          cacert
+          yq
         ];
 
         installPhase = ''
           runHook preInstall
 
+          lockfileVersion="$(yq -r .lockfileVersion pnpm-lock.yaml)"
+          if [[ ''${lockfileVersion:0:1} != ${lib.versions.major pnpm.version} ]]; then
+            echo "ERROR: lockfileVersion in pnpm-lock.yaml is not matching the provided pnpm major version!"
+            exit 1
```

Comment:
Some packages have older pnpm lockfile versions, but build just fine with pnpm_9. Maybe lockfile version mismatch should be only a warning?

An example is stylelint-lsp:

https://github.com/bmatcuk/stylelint-lsp/blob/3f9290571efd4f52bb1751587c9bce68fc182c16/pnpm-lock.yaml#L1

Reply:
🤔 I think we just need to check if the version is not greater than ours.